src/components/ModalExample.js

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import Modal from "./Modal";
+import useModal from "../services/hooks/useModal";
+
+export default function ModalExample() {
+  const { isShowing, toggle } = useModal();
+  const modalContent = [
+    { title: "Modal Title" },
+    { content: "Text" },
+    {
+      name: "Save", // if set to null this button will not be rendered
+      run: function () {
+        console.log("first btn clicked");
+      },
+    },
+    {
+      name: "Close",
+      run: function () {
+        console.log("second btn clicked");
+      },
+    },
+    {
+      name: "Info",
+      run: function () {
+        console.log("third btn clicked");
+      },
+    },
+  ];
+  return (
+    <>
+      <button className="button-default" onClick={toggle}>
```

Comment:
I know that this is just example but we are not going to use modal like that

Reply:
Roger